File: Haskell/euler_1.hs
-- Project Euler Question 1
-- Author: Corentin Pinato
-- Date: 24 May 2018
----------------------------------------------------------------
-- The prime factors of 13195 are 5, 7, 13 and 29.
-- What is the largest prime factor of the number 600851475143 ?
----------------------------------------------------------------

euler_1 :: Integer -> Integer
euler_1 n = sum $ filter (\x -> (x `mod` 3 == 0) || (x `mod` 5 == 0)) [1..n]

main = do
  print $ euler_1 1000

File: Python/euler_1.py
# Project Euler Question 1
# Author: Corentin Pinato
# Date: 24 May 2018
###############################################################################
# If we list all the natural numbers below 10 that are 
# multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
# Find the sum of all the multiples of 3 or 5 below 1000.
###############################################################################

num = 1000
numbers = [x for x in range(1, num)]
numbers = [x for x in numbers if ((x % 3) == 0 or (x % 5) == 0)]
print sum(numbers)
